Ldap/LdapManager.php

```diff
@@ -194,4 +207,60 @@ public static function unescapeValue($values = array())
 
         return (count($values) == 1 && array_key_exists(0, $values)) ? $values[0] : $values;
     }
+
+    /**
+     * Add roles based on role configuration from user memberOf attribute.
+     *
+     * @param UserInterface
+     * @param array $entry
+     */
+    private function addRolesFromMemberof($user, $entry)
+    {
+        if (isset($entry['memberof'])) {
+            foreach ($entry['memberof'] as $role) {
+                $dnSuffixFilter = $this->params['role']['memberOf']['dnSuffixFilter'];
+
+                if (preg_match("/^cn=(.*),$dnSuffixFilter/", $role, $roleName)) {
+                    $user->addRole(sprintf('ROLE_%s',
```

Comment:
This method needs to be documented. it's a requirement to have this method in a custom User class.

May add to the interface and also provide a trait for easy setup.